templates/openstackclient/bin/ansible-inventory

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+APISERVER=https://$KUBERNETES_SERVICE_HOST
+SERVICEACCOUNT=/var/run/secrets/kubernetes.io/serviceaccount
+NAMESPACE=$(cat ${SERVICEACCOUNT}/namespace)
+TOKEN=$(cat ${SERVICEACCOUNT}/token)
+CACERT=${SERVICEACCOUNT}/ca.crt
+
+ROLE_RESERVATIONS=$(curl -s --cacert ${CACERT} --header "Authorization: Bearer ${TOKEN}" -X GET ${APISERVER}/apis/osp-director.openstack.org/v1beta1/namespaces/$NAMESPACE/openstacknets/ctlplane | jq .spec.roleReservations | jq -r "to_entries | map(select(.value.addToPredictableIPs == true and .value.reservations[].vip == false and .value.reservations[].deleted == false)) | unique[]")
+
+ROLES=$(echo $ROLE_RESERVATIONS | jq -re '.key')
+
+# function prints out host IPs
+function printHostIPs {
+  local ROLE=$1
+  local ROLE_IPS=$2
+  local LAST_IP=$(echo $ROLE_IPS | sed -e "s|^.* ||")
+
+  for IP in $ROLE_IPS; do
+    if [[ "$IP" == "$LAST_IP" ]]; then
+      echo "        \"$IP\""
+    else
+      echo "        \"$IP\","
+    fi
+  done
+}
+
+
+if [ "$1" == "--list" ] ; then
+cat<<EOF
+{
+  "overcloud": {
+        "children": [
+EOF
+
+
+# print out the 'ALL' section (contains each rolename)
+LAST_ROLE=$(echo $ROLES | sed -e "s|^.* ||")
+for ROLE in $ROLES; do
+    if [[ "$ROLE" == "$LAST_ROLE" ]]; then
+      echo "        \"$ROLE\""
+    else
+      echo "        \"$ROLE\","
+    fi
+done
+echo '         ]'
+echo '  },'
+
+# print out the $ROLE sections which contain host IPs
+for ROLE in $ROLES; do
+ROLE_IPS=$(echo $ROLE_RESERVATIONS | jq -r "select(.key == \"$ROLE\") | .value.reservations[].ip")
+cat<<EOF
+  "$ROLE": {
+        "hosts": [
+EOF
+
+    printHostIPs "$ROLE" "$ROLE_IPS"
+
+echo '         ]'
```

Comment:
needs to be `echo '         ],'` as vars follows

Reply:
is it worth rewriting this in python or $(something that can serialise to json)?

Reply:
ack. I had already fixed this locally and just not pushed it yet. Still this functionally isn't working quite yet.